src/main/java/io/quarkus/gizmo/MethodCreator.java

```diff
@@ -57,4 +57,15 @@ default MethodCreator addException(Class<?> exception) {
 
     AnnotatedElement getParameterAnnotations(int param);
 
+    /**
+     * Sets names of all this method's parameters. The length of {@code parameterNames} must be equal
+     * to the number of this method's parameters. Does nothing if {@code parameterNames} is {@code null}.
```

Comment:
There are cases where the number of parameter names can or should be less than the number of parameters, e.g. inner class constructors. Not sure if that's too nitpicky but the constraint isn't completely correct.

Reply:
Synthetic and mandated parameters also have names, otherwise you can't match the name with the correct parameter. The enclosing instance parameter in case of inner class constructors are typically called `this$N` (where N >= 0).

Reply:
Technically what we could allow is omitting names for parameters at the _end_ of the parameter list. But that's hard to express without losing this check.

Or we could allow setting parameter names by position and synthesize names that were not set...

Reply:
I just realized I was wrong above. Synthetic and mandated parameters only have names in the local variable table. In the Parameters attribute, they do not.

I could perhaps check the number of parameter names against the signature, if present, but if not, I don't think this can meaningfully be checked. I guess I'll just have to remove this check for good and perhaps add some documentation.

Reply:
Added a little bit of documentation.

Reply:
I reviewed the specification and I revise my assessment. It says:

> The value of the parameters_count item indicates the number of parameter descriptors in the method descriptor ([§4.3.3](https://docs.oracle.com/javase/specs/jvms/se17/html/jvms-4.html#jvms-4.3.3)) referenced by the descriptor_index of the attribute's enclosing method_info structure.
>
> _This is not a constraint which a Java Virtual Machine implementation must enforce during format checking ([§4.8](https://docs.oracle.com/javase/specs/jvms/se17/html/jvms-4.html#jvms-4.8)). The task of matching parameter descriptors in a method descriptor against the items in the parameters array below is done by the reflection libraries of the Java SE Platform._

So I agree, the PR is OK as-is.